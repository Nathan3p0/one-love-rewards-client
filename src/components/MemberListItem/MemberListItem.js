import React from 'react'

const MemberListItem = (props) => {
    return (
        <li className="tableRow">
            <div className="col col-1" data-label="Created">{props.registered}</div>
            <div className="col col-2" data-label="Customer Name">{props.name}</div>
            <div className="col col-3" data-label="Customer Email">{props.email}</div>
            <div className="col col-4" data-label="Customer Phone">{props.phone}</div>
        </li>
    )
}

export default MemberListItem