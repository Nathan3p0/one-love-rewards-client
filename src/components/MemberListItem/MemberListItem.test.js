import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MemberListItem from './MemberListItem'

describe('Member List Item Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<MemberListItem />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<MemberListItem />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<MemberListItem
            registered={'2019-10-03T21:54:12.744Z'}
        />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})