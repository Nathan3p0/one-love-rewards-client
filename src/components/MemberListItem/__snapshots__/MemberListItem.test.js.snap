// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member List Item Component Rendering Tests renders to UI as expected 1`] = `
<li
  className="tableRow"
>
  <div
    className="col col-1"
    data-label="Created"
  >
    Invalid Date
  </div>
  <div
    className="col col-2"
    data-label="Customer Name"
  />
  <div
    className="col col-3"
    data-label="Customer Email"
  />
  <div
    className="col col-4"
    data-label="Customer Phone"
  />
</li>
`;

exports[`Member List Item Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<li
  className="tableRow"
>
  <div
    className="col col-1"
    data-label="Created"
  >
    10/3/2019
  </div>
  <div
    className="col col-2"
    data-label="Customer Name"
  />
  <div
    className="col col-3"
    data-label="Customer Email"
  />
  <div
    className="col col-4"
    data-label="Customer Phone"
  />
</li>
`;
